src/Reflection/EntityFieldReflection.php

```diff
@@ -41,6 +43,12 @@ public function getReadableType(): Type
             return new ObjectType($objectType);
         }
 
+        if ($this->propertyName === 'book'
+            && $this->declaringClass->is('Drupal\node\NodeInterface')
```

Comment:
Should this be `isSubclassOf`?

Reply:
@mglaman sorry for the delay in answering.

I actually tried this... and it didn't work (for my code anyway).

I'll try to explain as best I can why that is, but be aware (a) it was a few weeks ago so the details are a bit fuzzy and (b) I'm not an expert in how PHPStan understands things.

We had some functions that accepted NodeInterface as input, i.e.: `function foo(NodeInterface $a)` and others that accepted Node as input, i.e.: `function bar(Node $b)`.

Inside `function bar(Node $b)`, if we set `$b->bid = 123`, then `$this->declaringClass->isSubclassOf(...)` would return TRUE, i.e.: work just fine. But inside `function foo(NodeInterface $a)`, if we set `$a->bid = 123` then ``$this->declaringClass->isSubclassOf(...)` would return FALSE, i.e.: fail.

A bit of debugging suggests that `$this->declaringClass` would show up as a `NodeInterface`, and `isSubclassOf` would fail, I think this is because `\PHPStan\BetterReflection\Reflection\Adapter\ReflectionClass::isSubclassOf()` calls `getParentClassNames()` and it doesn't show up in that list.

Anyway, after much debugging, it seemed like `$this->declaringClass->is()` worked fine in both cases.